libs/ml/include/ml/ops/one_hot.hpp

```diff
@@ -0,0 +1,134 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "math/one_hot.hpp"
+#include "ml/ops/ops.hpp"
+
+#include <cassert>
+#include <utility>
+#include <vector>
+
+namespace fetch {
+namespace ml {
+namespace ops {
+
+template <class T>
+class OneHot : public fetch::ml::ops::Ops<T>
+{
+public:
+  using TensorType    = T;
+  using SizeType      = typename TensorType::SizeType;
+  using DataType      = typename TensorType::Type;
+  using ArrayPtrType  = std::shared_ptr<TensorType>;
+  using VecTensorType = typename Ops<T>::VecTensorType;
+  using SPType        = OpOneHotSaveableParams<T>;
+  using MyType        = OneHot<TensorType>;
+
+  explicit OneHot(SizeType depth, SizeType axis = 0, DataType on_value = DataType{1.0},
```

Comment:
please add a function comment that explains each of these parameters. I particularly find 'depth' to be an unintuitive parameter without some description